activesupport/lib/active_support/core_ext/integer/time.rb

```diff
@@ -38,7 +38,7 @@ def months
   alias :month :months
 
   def years
-    ActiveSupport::Duration.new(self * 365.25.days, [[:years, self]])
+    ActiveSupport::Duration.new(self * 365.25.days.to_i, [[:years, self]])
```

Comment:
`self` can't be a fraction here (we are extending `Integer`). Beyond that I wouldn't like to coerce results, just replace the constant result of `365.25.days` from _31557600.0_ to _31557600_